features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -1023,6 +1043,36 @@ static nsapi_error_t mbed_lwip_setsockopt(nsapi_stack_t *stack, nsapi_socket_t h
             }
             return 0;
 
+        case NSAPI_ADD_MEMBERSHIP:
+        case NSAPI_DROP_MEMBERSHIP: {
+#if LWIP_IPV6
+            return NSAPI_ERROR_UNSUPPORTED; 
+#else
+            if (optlen != sizeof(nsapi_ip_mreq_t)) {
+                return NSAPI_ERROR_UNSUPPORTED;
+            }
+
+            err_t igmp_err;
+            nsapi_ip_mreq_t *imr = (nsapi_ip_mreq_t *)optval;
+
+            ip_addr_t if_addr;
+            ip_addr_t multi_addr;
+
+            nsapi_addr_to_ip_addr(&multi_addr, &imr->imr_multiaddr);
+            nsapi_addr_to_ip_addr(&if_addr, &imr->imr_interface);
+
+            if (optname == NSAPI_ADD_MEMBERSHIP) {
+                igmp_err = igmp_joingroup(&if_addr, &multi_addr);
+            } else {
+                igmp_err = igmp_leavegroup(&if_addr, &multi_addr);
+            }
+            if (igmp_err != ERR_OK) {
```

Comment:
[It would default to NSAPI_ERROR_DEVICE_ERROR](https://github.com/ARMmbed/mbed-os/blob/master/features/FEATURE_LWIP/lwip-interface/lwip_stack.c#L690). If that's fine I can change it.